module/userExtension/helper.js

```diff
@@ -137,8 +143,16 @@ module.exports = class UserExtensionHelper {
                             entityMap[entity._id.toString()] = entity;
                         })
 
-                        for (let userExtensionRoleCounter = 0; userExtensionRoleCounter < userExtensionData[0].roles.length; userExtensionRoleCounter++) {
-                            for (let userExtenionRoleEntityCounter = 0; userExtenionRoleEntityCounter < userExtensionData[0].roles[userExtensionRoleCounter].entities.length; userExtenionRoleEntityCounter++) {
+                        for (
+                            let userExtensionRoleCounter = 0; 
+                            userExtensionRoleCounter < userExtensionData[0].roles.length; 
+                            userExtensionRoleCounter++
+                        ) {
```

Comment:
@aman-tunerlabs - This is already looping roles, why don't you generate ACL here itself

Reply:
Just a generics functionality i have created if i give roles i will get all the acl of that roles so that it can be reused when required later.

Reply:
Ok